DEV/org.openl.rules/src/org/openl/rules/dt/algorithm/DecisionTableAlgorithmBuilder.java

```diff
@@ -344,14 +345,27 @@ private boolean checkOtherColumnParametersInExpression(Condition condition) {
         return false;
     }
 
+    private static String toLowerCase(String str) {
+        if (str != null) {
+            return str.toLowerCase();
+        }
+        return null;
+    }
+
     private static boolean checkConditionParameterUsedInExpression(ICondition condition) {
         List<IdentifierNode> identifierNodes = DecisionTableUtils.retrieveIdentifierNodes(condition);
         for (IParameterDeclaration condParam : condition.getParams()) {
             if (identifierNodes.stream()
-                .anyMatch(identifierNode -> Objects.equals(condParam.getName(), identifierNode.getIdentifier()))) {
+                .anyMatch(identifierNode -> Objects.equals(toLowerCase(condParam.getName()),
+                    toLowerCase(identifierNode.getIdentifier())))) {

```

Comment:
1) toLowerCase(condParam.getName()) - extract outside of the cycle
2) String equalsIgnoreCase can be used on non-null string.

Reply:
String name = condParam.getName();

name != null && name.equalsIgnoreCase(identifierNode.getIdentifier())